sources/file.go

```diff
@@ -41,14 +41,21 @@ type File struct {
 	// outerPaths is the list of container paths (e.g. archives) that lead to
 	// this file
 	outerPaths []string
+
+	MaxArchiveDepth int
+
+	archiveDepth int
 }
 
 // Fragments yields fragments for the this source
 func (s *File) Fragments(yield FragmentsFunc) error {
 	ctx := context.Background()
 	format, _, err := archives.Identify(ctx, s.Path, nil)
-
 	if err == nil && format != nil {
+		if s.archiveDepth+1 > s.MaxArchiveDepth {
+			logging.Warn().Str("path", s.Path).Int("max archive depth", s.MaxArchiveDepth).Msg("max archive depth exceeded")
```

Comment:
```suggestion
			logging.Warn().Str("path", s.Path).Int("max_archive_depth", s.MaxArchiveDepth).Msg("max archive depth exceeded")
```

The slight tweak is because that will displayed as a variable in the logs (e.g. `max archive depth exceeded max_archive_depth=8`)